presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayIntersectFunction.java

```diff
@@ -14,69 +14,47 @@
 package com.facebook.presto.operator.scalar;
 
 import com.facebook.presto.operator.aggregation.TypedSet;
-import com.facebook.presto.spi.PageBuilder;
 import com.facebook.presto.spi.block.Block;
 import com.facebook.presto.spi.block.BlockBuilder;
 import com.facebook.presto.spi.function.Description;
 import com.facebook.presto.spi.function.ScalarFunction;
 import com.facebook.presto.spi.function.SqlType;
 import com.facebook.presto.spi.function.TypeParameter;
 import com.facebook.presto.spi.type.Type;
-import com.google.common.collect.ImmutableList;
 
 @ScalarFunction("array_intersect")
 @Description("Intersects elements of the two given arrays")
 public final class ArrayIntersectFunction
 {
-    private final PageBuilder pageBuilder;
-
-    @TypeParameter("E")
-    public ArrayIntersectFunction(@TypeParameter("E") Type elementType)
-    {
-        pageBuilder = new PageBuilder(ImmutableList.of(elementType));
-    }
+    private ArrayIntersectFunction() {}
 
     @TypeParameter("E")
     @SqlType("array(E)")
-    public Block intersect(
+    public static Block intersect(
             @TypeParameter("E") Type type,
             @SqlType("array(E)") Block leftArray,
             @SqlType("array(E)") Block rightArray)
     {
-        if (leftArray.getPositionCount() < rightArray.getPositionCount()) {
```

Comment:
This could still be helpful right? Or maybe the reverse logic. Maybe left should be the shorter one?

Reply:
Doesn't matter because we are going to actually traverse m+n elements anyway